### DISCUSSION TITLE
Implicit many-to-many relation saving multiple users whilst creating a game

### DISCUSSION THREAD

**Discussion by mah51** [prisma/prisma#9962] [Q&A]
Hi everyone, I am having some trouble creating game instances with 6 users in. The models are as follow
```
model User {
  id String @id
  avatar String 
  username String
  discriminator String
  games Game[]
}

model Game {
  id      Int   @id @default(autoincrement())
  players User[] 
  team1   String[]
  team2   String[]
  result  Int?

  createdAt DateTime @default(now()) @db.Timestamp(6)
}
```

When I try to create the game instance it wont let me create the array of Users... How do I connect multiple players to the game?
```
        const playerQueue = queue.map((user) => ({
            id: user.id,
            username: user.username,
            discriminator: user.discriminator,
            avatar: user.avatarURL()!
        }));

        const test = await this.container.prisma.game.create({
            data: {
                players: {
                    create: playerQueue
                },
                team1: playerQueue.splice(0, 2).map((usr) => usr.id),
                team2: playerQueue.splice(2, 5).map((usr) => usr.id)
            }
        });
```
As a note I get a typescript error when using connectOrCreate: playerQueue...

**Comment by austincrim** [comment]
Hey @mah51 👋 , thanks for your question.

Could you provide me the specific error you are receiving? Which line is it erroring on?

**Reply by mah51** [reply]
Hi thanks for the help,

The users arent saved into postsql:
![image](https://user-images.githubusercontent.com/47287285/138960408-5ad03a73-844c-44bb-8c58-e12ff2a3e62b.png)
![image](https://user-images.githubusercontent.com/47287285/138960427-6d407e3b-08d6-4f14-bc21-060c44997068.png)

and if I try using the createMany property instead I have to ts-ignore too: 
```
const test = await this.container.prisma.game.create({
            data: {
                players: {
                    createMany: { data: playerQueue }
                },
                team1: playerQueue.splice(0, 2).map((usr) => usr.id),
                team2: playerQueue.splice(2, 5).map((usr) => usr.id)
            }
        });
```

Unknown arg `createMany` in data.players.createMany for type UserCreateNestedManyWithoutGamesInput. Did you mean `create`? Available args:
```
type UserCreateNestedManyWithoutGamesInput {
   create?: UserCreateWithoutGamesInput | List<UserCreateWithoutGamesInput> | UserUncheckedCreateWithoutGamesInput | List<UserUncheckedCreateWithoutGamesInput>        
    connectOrCreate?: UserCreateOrConnectWithoutGamesInput | List<UserCreateOrConnectWithoutGamesInput>
    connect?: UserWhereUniqueInput | List<UserWhereUniqueInput>
}
```

**Comment by austincrim** [comment]
I can successfully get `players` inserted using this snippet:
```ts
let players = [
    {
      id: '123',
      username: 'austin',
      discriminator: 'foo',
      avatar: 'avatar.jpeg'
    },
    {
      id: '456',
      username: 'michael',
      discriminator: 'bar',
      avatar: 'avatar.png'
    }
  ]
  await prisma.game.create({
    data: {
      players: {
        create: players
      }
    }
  })
```

![image](https://user-images.githubusercontent.com/32459922/138964397-1da6ed9b-3820-400d-84ca-28d6e4adbdec.png)

Are you certain that all the correct fields are coming through on `playerQueue`? 
Also, would you mind telling me your `prisma` version?

**Reply by mah51** [reply]
Thanks @austincrim , I have tried your code and it also works for me, however whilst saving other properties on the game it does not create the users eg:
```
let players = [
			{
				id: '123',
				username: 'austin',
				discriminator: 'foo',
				avatar: 'avatar.jpeg'
			},
			{
				id: '456',
				username: 'michael',
				discriminator: 'bar',
				avatar: 'avatar.png'
			}
		];
		await prisma.game
			.create({
				data: {
					players: {
						create: players
					},
					team1: players.splice(0, 2).map((usr) => usr.id),
					team2: players.splice(2, 5).map((usr) => usr.id)
				}
			})
```
which results in the same outcome as before where the game is created but users are not created and not related to the game.

Prisma and @prisma/client are v3.3.0

**Reply by austincrim** [reply]
I think you want to use `players.slice` instead of `players.splice`. `splice` will modify the original array and is likely causing some issues.

**Reply by austincrim** [reply]
👋 @mah51, just following up, did you get your issue resolved?

**Reply by mah51** [reply]
Very sorry, I forgot to reply! Yes this solved my issue. Thanks a lot Austin for your time.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
